hpx/parallel/datapar/detail/iterator_helpers.hpp

```diff
@@ -0,0 +1,312 @@
+
+                std::advance(it, V1::Size);
+                std::advance(dest, ret.size());
+            }
+
+            template <typename F, typename InIter1, typename InIter2,
+                typename OutIter>
+            HPX_HOST_DEVICE HPX_FORCEINLINE
+            static void call1(F && f, InIter1& it1, InIter2& it2, OutIter& dest)
+            {
+                typedef typename std::iterator_traits<InIter1>::value_type
+                    value1_type;
+                typedef typename std::iterator_traits<InIter2>::value_type
+                    value2_type;
+
+                typedef Vc::Scalar::Vector<value1_type> V1;
+                typedef Vc::Scalar::Vector<value2_type> V2;
+
+                V1 tmp1(std::addressof(*it1), Vc::Aligned);
+                V2 tmp2(std::addressof(*it2), Vc::Aligned);
+                auto ret = hpx::util::invoke(f, &tmp1, &tmp2);
+                ret.store(std::addressof(*dest), Vc::Aligned);
+
+                std::advance(it1, V1::Size);
+                std::advance(it2, V2::Size);
+                std::advance(dest, ret.size());
+            }
+
+            ///////////////////////////////////////////////////////////////////
+            template <typename F, typename InIter, typename OutIter>
+            HPX_HOST_DEVICE HPX_FORCEINLINE
+            static void callv(F && f, InIter& it, OutIter& dest)
+            {
+                typedef typename std::iterator_traits<InIter>::value_type
+                    value_type;
+
+                typedef Vc::Vector<value_type> V;
+
+                if (data_alignment(it) || data_alignment(dest))
+                {
+                    V tmp(std::addressof(*it), Vc::Unaligned);
+                    auto ret = hpx::util::invoke(f, &tmp);
+                    ret.store(std::addressof(*dest), Vc::Unaligned);
+                    std::advance(dest, ret.size());
+                }
+                else
+                {
+                    V tmp(std::addressof(*it), Vc::Aligned);
+                    auto ret = hpx::util::invoke(f, &tmp);
+                    ret.store(std::addressof(*dest), Vc::Aligned);
+                    std::advance(dest, ret.size());
+                }
+
+                std::advance(it, V::Size);
+            }
+
+            template <typename F, typename InIter1, typename InIter2,
+                typename OutIter>
+            HPX_HOST_DEVICE HPX_FORCEINLINE
+            static void callv(F && f, InIter1& it1, InIter2& it2, OutIter& dest)
+            {
+                typedef typename std::iterator_traits<InIter1>::value_type
+                    value1_type;
+                typedef typename std::iterator_traits<InIter2>::value_type
+                    value2_type;
+
+                typedef Vc::Vector<value1_type> V1;
+                typedef Vc::Vector<value2_type> V2;
+
+                if (data_alignment(it1) || data_alignment(it2) ||
+                    data_alignment(dest))
```

Comment:
I'm not sure if its worth adding yet another code path for this case. But it could be done for sure.